src/App.tsx

```diff
@@ -8,53 +10,179 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import {
+  addNewComment,
+  deleteComment,
+  getUserPostComments,
+  getUserPosts,
+  getUsers,
+} from './api/post';
+import { Post } from './types/Post';
+import { Comment, CommentData } from './types/Comment';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userSelected, setUserSelected] = useState<User | null>(null);
+  const [showPostLoader, setShowPostLoader] = useState<boolean>(false);
+  const [showCommentLoader, setShowCommentLoader] = useState<boolean>(false);
+  const [isAddingComment, setIsAddingComment] = useState(false);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isSidebarVisible, setIsSidebarVisible] = useState<number | null>(null);
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [hasPostError, setHasPostError] = useState<boolean>(false);
+  const [hasCommentError, setHasCommentError] = useState<boolean>(false);
+  const [isCommentFormVisible, setIsCommentFormVisible] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUsers()
+      .then(res => {
+        setUsers(res as User[]);
+      })
+      .catch(() => new Error('Failed to fetch users'));
+  }, []);
```

Comment:
It's a good practice to handle errors properly. In your case, if the request fails, you just throw a new Error, but nothing catches it. It's better to show some notification to the user about the problem. You can use a try-catch block for error handling.